frontend/src/components/Calendar/Calendar.js

```diff
@@ -24,19 +39,92 @@ const Calendar = () => {
     setCurrentEvent(null);
   };
 
-  // Fetches events from API
+  // Function to handle states when an update form is opened
+  const openUpdateForm = () => {
+    setCurrentUpdate(currentEvent);
+    setIsUpdateOpen(true);
+    console.log("Opening update form for event:", currentEvent.title);
+  }
+
+  // Function to handle states when an update form is closed
+  const closeUpdateForm = () => {
+    console.log("Closing update form for event:", currentUpdate.title);
+    setCurrentUpdate(null);
+    setIsUpdateOpen(false);
+  }
+
+  // Function to live update (no browser refresh required) the calendar when an event is updated 
+  const updateEvent = (updatedEvent) => {
+    setEvents((previousEvents) => previousEvents.map((event) => event.id === updatedEvent.id ? updatedEvent : event));
+  };
+
+  // Function to handle states when a create form is opened
+  const openCreateForm = () => {
+    setIsCreateOpen(true);
+    console.log("Opening create form.");
+  }
+
+  // Function to handle states when a create form is closed
+  const closeCreateForm = () => {
+    setIsCreateOpen(false);
+    console.log("Closing create form");
+  } 
+
+  // Function to display a newly created event onto the calendar locally without refreshing the browser
+  const createEvent = (newEvent) => {
+    setEvents((prevEvents) => [...prevEvents, newEvent]); 
+    console.log("Created Event: ", newEvent.title);
+  }
+
+  // Function to delete the current event on a modal
+  const deleteEvent = async (deletedEvent) => {
+    try {
+      // Make DELETE request to delete the event
+      await api.delete(`${route}/${deletedEvent.id}/delete/`);
+      setEvents((previousEvents) => previousEvents.filter((event) => event.id !== deletedEvent.id));
+      closeModal();
+    } catch (error) {
+      console.log("Error deleting event: ", error.response);
+    } 
+  }
+
+  // Fetches events and hubs from API
   useEffect(() => {
-    console.log("Fetching...");
+
+    // Need to fix this. Works if first time trying to access website, but doesn't work for subsequent uses.
+    const token = localStorage.getItem(ACCESS_TOKEN);
+    if(!token){
+      console.log("User is currently not logged in. Redirecting to login page.");
+      router.push('/users/login');
+      return;
+    }
+
+    console.log("Fetching the hubs the current user is a owner of or is moderating...");
+    const fetchHubs = async () => {
+      try {
+          const response = await api.get("/api/hubs/modding/", {headers: {Authorization: `Bearer ${token}`}});
```

Comment:
I don't think specifying the authorization token is necessary when using our api utility when in a ProtectedRoute. For example, the page that displays a post list currently uses this: 

![image](https://github.com/user-attachments/assets/e92373cc-f90b-47a8-97ca-bc8ca513bbbd)

Maybe I'm overlooking something, but I thought the api component added the token by itself.

Reply:
You are right about this. I removed the authorization token and the output is the same. I will remove it and update it.